core/src/main/java/com/linecorp/armeria/common/util/AbstractThreadFactory.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2019 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common.util;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.concurrent.ThreadFactory;
+import java.util.function.Function;
+
+import javax.annotation.Nullable;
+
+import io.netty.util.concurrent.DefaultThreadFactory;
+
+/**
+ * A skeletal {@link ThreadFactory} implementation.
+ */
+public abstract class AbstractThreadFactory implements ThreadFactory {
+    // Note that we did not extend DefaultThreadFactory directly to hide it from the class hierarchy.
+    private final ThreadFactoryImpl delegate;
+    private Function<? super Runnable, ? extends Runnable> taskFunction = Function.identity();
+
+    protected AbstractThreadFactory(String threadNamePrefix) {
+        delegate = new ThreadFactoryImpl(requireNonNull(threadNamePrefix, "threadNamePrefix"));
+    }
+
+    protected AbstractThreadFactory(String threadNamePrefix, boolean daemon) {
+        delegate = new ThreadFactoryImpl(requireNonNull(threadNamePrefix, "threadNamePrefix"),
+                                              daemon);
+    }
+
+    protected AbstractThreadFactory(String threadNamePrefix, int priority) {
+        delegate = new ThreadFactoryImpl(requireNonNull(threadNamePrefix, "threadNamePrefix"),
+                                              priority);
+    }
+
+    protected AbstractThreadFactory(String threadNamePrefix, boolean daemon, int priority) {
+        delegate = new ThreadFactoryImpl(requireNonNull(threadNamePrefix, "threadNamePrefix"),
+                                              daemon, priority);
+    }
+
+    protected AbstractThreadFactory(String threadNamePrefix, boolean daemon, int priority,
+                                    @Nullable ThreadGroup threadGroup) {
+        delegate = new ThreadFactoryImpl(requireNonNull(threadNamePrefix, "threadNamePrefix"),
+                                              daemon, priority, threadGroup);
+    }
+
+    protected AbstractThreadFactory(String threadNamePrefix, boolean daemon, int priority,
```

Comment:
This is a public class so I think we should deprecate all the constructors above except this one.

Reply:
Fixed it. Thank you.